src/electron/platform/android/setup/live-android-setup-deps.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { UserConfigurationStore } from 'background/stores/global/user-configuration-store';
+import { Logger } from 'common/logging/logger';
+import { compareSemverValues, SemverComparisonResult } from 'electron/common/semver-comparer';
+import { AndroidServiceApkLocator } from 'electron/platform/android/android-service-apk-locator';
+import {
+    AndroidServiceConfigurator,
+    AndroidServiceConfiguratorFactory,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+import { AndroidSetupDeps } from 'electron/platform/android/setup/android-setup-deps';
+
+export class LiveAndroidSetupDeps implements AndroidSetupDeps {
+    private serviceConfig: AndroidServiceConfigurator;
+    private selectedDeviceId: string;
+
+    constructor(
+        private readonly configFactory: AndroidServiceConfiguratorFactory,
+        private readonly configStore: UserConfigurationStore,
+        private readonly apkLocator: AndroidServiceApkLocator,
+        private readonly logger: Logger,
+    ) {}
+
+    public hasAdbPath = async (): Promise<boolean> => {
+        try {
+            const adbLocation = this.configStore.getState().adbLocation;
+            this.serviceConfig = await this.configFactory.getServiceConfigurator(adbLocation);
+            return true;
+        } catch (error) {
+            this.logger.log(error);
+            return false;
+        }
+    };
+
+    public setAdbPath = (path: string): void => {
+        this.configStore.getState().adbLocation = path;
+    };
+
+    public getDevices = async (): Promise<DeviceInfo[]> => {
+        return await this.serviceConfig.getConnectedDevices();
```

Comment:
I believe you don't need to await this promise. As it is, this function isn't really async.